[project]
name = "mpt-finops-operations"
version = "0.1.0"
description = "Add your description here"
readme = "README.md"
requires-python = ">=3.12"
dependencies = [
    "alembic>=1.14.0",
    "asyncpg>=0.30.0",
    "fastapi-async-sqlalchemy>=0.6.1",
    "fastapi-camelcase>=2.0.0",
    "fastapi-pagination>=0.12.32",
    "fastapi[standard]>=0.115.5",
    "pycountry>=24.6.1",
    "pydantic-extra-types>=2.10.0",
    "pydantic-settings>=2.6.1",
    "python-dotenv>=1.0.1",
    "sqlalchemy[asyncio]>=2.0.36",
    "sqlmodel>=0.0.22",
]

[dependency-groups]
dev = [
    "bandit>=1.8.0",
    "httpx>=0.27.2",
    "ipython>=8.30.0",
    "pytest>=8.3.3",
    "pytest-asyncio>=0.24.0",
    "pytest-cov>=6.0.0",
    "ruff>=0.8.0",
    "typer>=0.13.1",
]


[tool.ruff]
line-length = 100
target-version = "py312"
output-format = "full"
extend-exclude = [
    "migrations/versions/",
]

[tool.ruff.lint]
select = [
    "E",  # w errors
    "W",  # pycodestyle warnings
    "F",  # pyflakes
    "I",  # isort
    "B",  # flake8-bugbear
    "C4",  # flake8-comprehensions
    "UP",  # pyupgrade,
    "PT",  # flake8-pytest-style
    "T10",  # flake8-pytest-style
]
ignore = [
    "PT001", # Use `@pytest.fixture()` over `@pytest.fixture`
    "PT011", # pytest.raises({exception}) is too broad, set the match parameter or use a more specific exception
    "PT023", # @pytest.mark.parametrize` is missing required arguments
    "B008",  # do not perform function calls in argument defaults
    "B904",  # Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling
]

[tool.ruff.format]
quote-style = "double"
docstring-code-format = true

[tool.bandit]
exclude_dirs = ["tests", ".venv"]

[tool.pytest.ini_options]
testpaths = "tests/"
pythonpath = "."
asyncio_mode = "auto"
asyncio_default_fixture_loop_scope = "session"
addopts = """
    -vvv \
    --cov --cov-report html \
"""

[tool.coverage.run]
branch = true
source = ["app"]

[tool.coverage.report]
show_missing = true
fail_under = 50
